internal/adsysservice/adsysservice.go

```diff
@@ -178,29 +170,34 @@ func New(ctx context.Context, url, domain string, opts ...option) (s *Service, e
 		return nil, err
 	}
 
-	url, domain, defaultDomainSuffix, err := loadServerInfo(args.sssdConf, url, domain, args.defaultDomainSuffix)
-	if err != nil {
-		return nil, err
-	}
-
 	var adOptions []ad.Option
 	if args.cacheDir != "" {
 		adOptions = append(adOptions, ad.WithCacheDir(args.cacheDir))
 	}
 	if args.runDir != "" {
 		adOptions = append(adOptions, ad.WithRunDir(args.runDir))
 	}
-	if args.sssCacheDir != "" {
-		adOptions = append(adOptions, ad.WithSSSCacheDir(args.sssCacheDir))
+
+	// AD Backend selection
+	var adBackend backends.Backend
+	switch args.adBackend {
+	default:
+		log.Warningf(ctx, "Unknown configured backend %q. Defaulting to sssd.", args.adBackend)
+		fallthrough
+	case "":
+		fallthrough
+	case "sssd":
+		adBackend, err = sss.New(ctx, args.sssConfig, bus)
+	case "winbind":
 	}
-	if defaultDomainSuffix != "" {
-		adOptions = append(adOptions, ad.WithDefaultDomainSuffix(defaultDomainSuffix))
+	if err != nil {
+		return nil, fmt.Errorf(i18n.G("could not initialize AD backend: %s"), err)
```

Comment:
Also would be better to wrap the error with %w rather than using its string value (%s)

Reply:
see my other comment on wrapping.

However, it should be `%v` here and not `%s`, changing this, thanks!